lib/generators/smashing_documentation/install_generator.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails/generators'
+module SmashingDocumentation
+  module Generators
+    class InstallGenerator < Rails::Generators::Base
+      source_root File.expand_path("../../../templates/", __FILE__)
+
+      def install
+        if Dir.exists?('spec')
+          update_rails_helper
+          update_spec_helper
+          generate_docs_template
+        else
+          puts "It does not appear that you have RSpec installed in your app"
+          puts "Please set up RSpec before running this installer"
+        end
+      end
+
+      private
+
+      def update_rails_helper
+        destination = "spec/rails_helper.rb"
+        create_file(destination) unless File.exist?(destination)
+        append_file(destination,
+          "SmashingDocs.config do |c|\n"\
+          "  c.template_file = 'smashing_docs/template.md'\n"\
+          "  c.output_file   = 'smashing_docs/api_docs.md'\n"\
+          "end"
+          ) unless File.readlines(destination).grep(/SmashingDocs.config/).any?
+      end
+
+      def update_spec_helper
+        destination = "spec/spec_helper.rb"
+        if File.exist?(destination)
+          insert_into_file(
+            destination,
+            "\n  config.after(:each, type: :controller) do\n"\
+            "    SmashingDocs.run!(request, response)\n"\
+            "  end\n"\
+            "  config.after(:suite) { SmashingDocs.finish! }",
```

Comment:
For these code snippets, it might be a good idea to put them in their own files, and read them in a string here.